code/modules/projectiles/pins.dm

```diff
@@ -134,9 +134,12 @@
 
 // Ultra-honk pin, clown's deadly joke item.
 // A gun with ultra-honk pin is useful for clown and useless for everyone else.
+/obj/item/firing_pin/clown/ultra
+	name = "ultra hilarious firing pin"
+
 /obj/item/firing_pin/clown/ultra/pin_auth(mob/living/user)
 	playsound(src.loc, 'sound/items/bikehorn.ogg', 50, TRUE)
-	if(user && (!(HAS_TRAIT(user, TRAIT_CLUMSY)) && !(user.mind && user.mind.assigned_role == "Clown")))
+	if(user && (!(HAS_TRAIT(user, TRAIT_CLUMSY)) && !(user.mind && user.mind.assigned_role == "Clown") && !(user.mind && user.mind.has_antag_datum(/datum/antagonist/nukeop/clownop)) && !(user.mind && user.mind.has_antag_datum(/datum/antagonist/nukeop/leader/clownop)))) //Yep, the antag datum for clown op leaders isn't a subtype of the antag datum for normal clown ops.
```

Comment:
This could be broken onto several checks for readability.
```nix
if(!user)
  return TRUE
if(HAS_TRAIT(user, TRAIT_CLUMSY))
  return TRUE
```
And so on.